### ISSUE TITLE
Benchmark design questions

### ISSUE THREAD

**Issue by hongshuh** [janosh/matbench-discovery#39]
I'm having a concern regarding the MBD benchmark. The protocol suggests training on the MP datasets and testing on the WBM dataset. Could you please confirm if "energy above hull" is the correct training label and if "energy_above_hull_mp2020_corrected_ppd_mp" is the appropriate ground truth label for the test set? I've observed a noticeable difference in the distribution between the training and testing datasets(there will be no negative value in the train set). Moreover, each dataset seems to imbalance between "stable" and "unstable" classifications. Do you think mixing the two datasets and then performing a 5-fold cross-validation will be better, similar to the process used in Matbench?

**Comment by CompRhys** [comment]
The protocol is to train on the formation energy (or if you prefer you could train on the vasp energy) our analysis code can then calculate what the predicted energy above the hull (EACH) would be given your prediction of the formation energy from which the metrics of interest (DAF/F1/Precision/Recall) can be determined.

The distributional difference is intended as it matches prospective applications. In the training set there by definition cannot be anything below the convex hull of the training set. The aim of a discovery workflow is to identify structures that may be below the convex hull of known materials. As a result our test set must contain materials that lie below the convex hull of the training set in order to usefully test this application. 

Matbench is a great effort but it doesn't test the same extrapolatory insights as our proposed evaluation framework. Mixing the WBM and MP data sets and following a Matbench-esque CV setup would not provide any more insight than might be obtained from the Matbench formation energy task that already exists. The reason we have proposed the matbench discovery evaluation framework is that we believe that it provides uniquely useful insights for a real application that are missing from more conventional IID testing setups.

**Comment by hongshuh** [comment]
I see, that makes sense, is there any way to deal with the imbalance? Predicting every material as "unstable" can reach an accuracy of 83%, only CHGNet is slightly above.

**Comment by hongshuh** [comment]
Could you remind me which code is used to calculate the predicted energy above the convex hull from the prediction of formation energy?

**Comment by janosh** [comment]
@hongshuh For the most, we don't need to compute actual hull distances. Have a look at https://matbench-discovery.materialsproject.org/si#formation-energy-mae--hull-distance-mae for an explanation.

**Comment by CompRhys** [comment]
There's no class imbalance if you frame the problem as a regression for the formation energies.

The DAF of your proposed baseline is 0. This is why we were careful to select metrics that actually test what we're interested in - machine learning models that accelerate discovery.

**Comment by hongshuh** [comment]
Thanks for your reply! It will be a great benchmark for materials discovery

**Comment by janosh** [comment]
@hongshuh Thanks for your interest and feedback. We certainly hope so. If you'd like to get involved, let us know. Since this is not a conventional paper, we can add whatever analysis we like whenever. All the model predictions are in this repo. If you'd like to dig in and suggest new metrics or visualizations that reveal useful insights into what ML can and can't do in the context of materials discovery, you're very welcome!

**Comment by hongshuh** [comment]
And one of the  insterested point is mentioned by @CompRhys , I would like to know how to compared the structured-based or UIP models with coordinate-free. Even though CHGNet is the current SOTA model, it requires considerably more information compared to the coordinate-free model like Wreformer. I am not an expert in DFT, so I am wondering the cost of generating the relaxed structure data. If the cost is high, then the Wrenformer will be the prefered one even the DAF is lower, It would be beneficial to have a metrics to represent the efficency.

**Comment by janosh** [comment]
You mean efficiency to generate the training data? That'll be very hard to quantify, I'm afraid.

I think such analysis would be somewhat relevant but I expect most people would still select UIPs if the trade-off is 1000x faster than DFT and 4x the hit rate, or 100,000x faster and 2x the hit rate for a coordinate-free model like Wrenformer.

**Comment by hongshuh** [comment]
I mean for discovery the potential materials, We can easily generates millions of formula and space group (those are just strings) in minutes, how abouts those UIPs?

**Comment by janosh** [comment]
The thing is that training data already exists openly and is growing. So people don't mind the original cost to generate it. They just care about the best model you can train right now.

**Comment by CompRhys** [comment]
In practice I expect that the most likely way people will use UIP for discovery is via a prototype based approach at least until we've got enough data in the field to train "foundation" models that are generally trusted for GA/RSS/SA methods. As such the choice of UIP vs Wrenformer is whether you just look at the prototype to make a prediction then relax with DFT or whether you perform a cheap UIP relaxation then make a prediction before a final DFT relaxation if cheaper. As Janosh said this is about 100x slower but still fast with the time per material being measured in seconds not minutes. As such if UIPs are the best according to our benchmark then even as the author of Wren upon which Janosh's Wrenformer variant is based UIP may be better suited to this type of campaign.

**Comment by CompRhys** [comment]
All you need to do for the benchmark is predict E_f we pre calculate everything else. This is intentional to reduce the surface area where people might make mistakes.

That said if you have explicit errors feel free to post the trace here (in another issue for clarity) or better yet in `pymatgen`.

**Comment by janosh** [comment]
@hongshuh Like Rhys said, just send us your formation energy preds. However, in case you're still interested, here's the specific line using the `PatchedPhaseDiagram`:

https://github.com/janosh/matbench-discovery/blob/297251c0f24bc2f62e6fb5a6ab0a1883cff29e9e/data/wbm/fetch_process_wbm_dataset.py#L559
